### ISSUE TITLE
Editing Time Tracking details behaves unpredictable

### ISSUE THREAD

**Issue by Akjo4712** [invoiceninja/admin-portal#276]
**What version of Invoice Ninja are you running? ie v4.5.25 / v5.0.30**
v5.1.65-C48

**What environment are you running?**
Docker

**Have you checked log files (storage/logs/) Please provide redacted output**
No

**Have you searched existing issues?**
Yes

**Have you reported this to Slack/forum before posting?**
No

**Describe the bug**
When editing or adding Time Tracking Entries, sometimes it is overwriting and or deleting previous Entries, which is critical if we use Invoice Ninja as primary Time Tracking System for Billing. 

When I am affected the whole Time Tracking Screen is behaving weird.

As i am unable to write steps to reproduce or to describe the Issue, I will take a short Video and Post it in the Comments as soon as I can reproduce the Issue again.

**Steps To Reproduce**
Unable to provide "Steps To Reproduce", see Video in Comments.

**Expected behavior**
Time Tracking should only delete/edit the Entries I directly interact with.

**Screenshots**
Unable to provide "Screenshots", see Video in Comments.

**(v5) Can you replicate the issue on our demo site? https://demo.invoiceninja.com**
Did invest some minutes and tried to reproduce on my instance and on the demo, but i am unable for now on both.

**Comment by hillelcoren** [comment]
Hi,

Can you see any pattern which triggers the problem?

**Comment by Akjo4712** [comment]
I was just able to reproduce the Issue on my Docker Instance, but I had no Screen Recording running. Is someone else seeing unpredictable behavior when filling out details in this Screen?

Deleting the last Row resulted in another Row beeing generated automatically. When deleting this Row, then always the Row above got deleted. Just as there is some array index which gets mixed up.

Could be related with Rows which are not filled out completly. I will take a look if I can trace it down.

![image](https://user-images.githubusercontent.com/65533611/119947412-fe290480-bf97-11eb-9b25-2afc7576d774.png)

**Comment by Akjo4712** [comment]
@hillelcoren, Last time i was able to reproduce the Issue, I did following steps. Task with 13 Time Tracking Entries, see Screenshot above.

1. Open Edit Screen
2. Add some Rows with just a Startdate
3. Try deleting the Rows

- Deleting one Row Adds Starttime to all Entries which are missing one.
- Also Deleting one of the Rows results in the Row above beeing removed.
- When deleting the last so added Row, you need to click on delete more than once.

**Comment by hillelcoren** [comment]
The behavior you're describing sounds like "Auto start tasks" may be enabled on Settings > Task Settings

**Comment by hillelcoren** [comment]
It's very possible there is a problem. 

If you can provide specific steps to replicate the problem it would be very helpful.

**Comment by Akjo4712** [comment]
"Auto start tasks" is diabled. I will try to find the specific steps to replicate the problem. Just give me some days, I will get back to you.

**Comment by hillelcoren** [comment]
Note: although the UI shows date/start/end/duration the app stores the start and end date/times. This means if you (for example) set the start time the start date will be set for you automatically. These automatic values may not be visible right away which may explain the behaviour you're seeing.

**Comment by Akjo4712** [comment]
I was just able to Record one of the unpredictable things. Might I send you the Video on another channel then Github? Therer is Customer Data visible, which I cant't remove from the Video.

 What happend:
1. I opende an existig Task with existing Time Tracking Details
2. I added a Row in the Time Tracking Details
3. This automatically added a new Row which I don't need
4. I had to click 3 Times on the Delete Button until the Row is actually deleted

**Comment by hillelcoren** [comment]
If the video shows what you described I'm not sure if it will be helpful.

Do you always see this behavior/do you see a pattern as to when it occurs? 

If you can provide replicable steps we should be able to correct it.

**Comment by hillelcoren** [comment]
To clarify... the problem is that you need to click multiple times to delete a row?

Above you wrote:
> I added a Row in the Time Tracking Details
> This automatically added a new Row which I don't need

Do you mean it added two rows?

**Comment by Akjo4712** [comment]
Yes, it added two rows. When i enter a Start Time for my new Entry i get an additional Row. In order to delete the additional Row which I never wanted, i have to click the delete Button 3 Times.

I have the Video ready to send to you, just would need a privat channel, so i don't leak customer information.

**Comment by hillelcoren** [comment]
Does this also happen when creating a new task?

**Comment by Akjo4712** [comment]
Yes, this also is happening for new and not saved Tasks.

While I am typing the Start Date for my Row I suddenly get another Row with the same Start Date. If I would like to delete the additional new Row, which I don't need, I have to click on delete for 3 Times until it get's deleted.

I just also found out, that I can reproduce the 3 clicks to delete if i just enter a Start Date for a new Row and if I then try to delete it.

With the first click i can see a Start Time, with the second click nothing changes, with the third click the entry get's deleted.

**Comment by hillelcoren** [comment]
Can you reproduce it here: https://demo.invoiceninja.com

**Comment by Akjo4712** [comment]
Yes I can reproduce on https://demo.invoiceninja.com, but only after changing the Date Format in the Localisation Settings to "31.01.2000". Might this be the missing Part in the Puzzle?

**Comment by hillelcoren** [comment]
I still can't reproduce, can you post a video of the issue recorded on demo?

**Comment by Akjo4712** [comment]
Of course. https://1drv.ms/v/s!AuTKro0jFwa-heUDcc8DAybvIKpZAw?e=SjBLuF

**Comment by hillelcoren** [comment]
Thanks, I'm able to reproduce the problem

**Comment by Akjo4712** [comment]
Nice! As soon as this one is sorted out, I will check if the other unpredictable behaviours where just side effects of this Issue.

**Comment by hillelcoren** [comment]
I've made a change which should help

**Comment by Akjo4712** [comment]
Is this included in 5.1.72? If so I would make some extended testing in order to see if the other unpredictable things I noticed just where causes of this Issue.

**Comment by hillelcoren** [comment]
Yes, this change is included
